components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/soaptorest/template/RESTToSOAPMsgTemplate.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.apimgt.impl.soaptorest.template;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.velocity.Template;
+import org.apache.velocity.VelocityContext;
+import org.apache.velocity.app.VelocityEngine;
+import org.apache.velocity.runtime.RuntimeConstants;
+import org.json.simple.JSONArray;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.APIManagerConfigurationService;
+import org.wso2.carbon.apimgt.impl.internal.ServiceReferenceHolder;
+import org.wso2.carbon.apimgt.impl.template.ConfigContext;
+
+import java.io.File;
+import java.io.StringWriter;
+import java.util.Map;
+
+/**
+ *  velocity template to generate in sequences and out sequences
+ */
+public class RESTToSOAPMsgTemplate {
+
+    private static final Log log = LogFactory.getLog(RESTToSOAPMsgTemplate.class);
+
+    private String velocityLogPath = null;
+    private static final String IN_SEQ_TEMPLATE_FILE = "soap_to_rest_in_seq_template";
+    private static final String OUT_SEQ_TEMPLATE_FILE = "soap_to_rest_out_seq_template";
+
+    /**
+     * gets in sequence for the soap operation
+     *
+     * @param mapping soap to rest mapping json
+     * @param method http method for the resource
+     * @param soapAction soap action for the operation
+     * @param namespace soap namespace for the sequence
+     * @return in sequence string
+     */
+    public String getMappingInSequence( Map<String, String> mapping, String method, String soapAction, String namespace, JSONArray array) {
+
+        ConfigContext configcontext = new SOAPToRESTConfigContext(mapping, method, soapAction, namespace, array);
+        StringWriter writer = new StringWriter();
+        try {
+            VelocityContext context = configcontext.getContext();
+            context.internalGetKeys();
+
+            VelocityEngine velocityengine = new VelocityEngine();
+            if (!"not-defined".equalsIgnoreCase(getVelocityLogger())) {
+                velocityengine.setProperty( RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+                        "org.apache.velocity.runtime.log.Log4JLogChute" );
+                velocityengine.setProperty( "runtime.log.logsystem.log4j.logger", getVelocityLogger());
+            }
```

Comment:
Shall we use the constants?

Reply:
fixed in e18f43f5dee9ed43a3239af62f38d67c72c78fa2